gamemodes/horde/gamemode/sv_rank.lua

```diff
@@ -142,28 +142,28 @@ function HORDE:LoadRank(ply)
 end
 
 local ExpMultiConvar = GetConVar("horde_experience_multiplier")
+local StartXPMult = Horde.Difficulty[Horde.CurrentDifficulty].xpMultiStart
+local EndXPMult = Horde.Difficulty[Horde.CurrentDifficulty].xpMultiEnd
 
 if GetConVar("horde_enable_sandbox"):GetInt() == 0 and GetConVar("horde_enable_rank"):GetInt() == 1 then
 	hook.Add("Horde_OnEnemyKilled", "Horde_GiveExp", function(victim, killer, wpn)
-
-		local ExpMulti = ExpMultiConvar:GetInt()
 			
-		if HORDE.CurrentDifficulty >= 5 and HORDE.current_wave >= 4 then
-			ExpMulti = 2 * ExpMulti
-		end
+		local WavePercent = Horde.current_wave / 10 --If we ever change the max # of waves this will have to become dynamic instead of 10
+		local RoundXPMulti = StartXPMult+(WavePercent*(EndXPMult-StartXPMult)) --This gets the xp multi number between min and max multi based on round
+		local ExpMulti = RoundXPMulti * ExpMultiConvar:GetInt()
 
 		if HORDE.current_wave <= 0 or GetConVar("sv_cheats"):GetInt() == 1 then return end
 		if killer:IsPlayer() and killer:IsValid() and killer:Horde_GetClass() then
```

Comment:
shouldathis should return early

Reply:
you mean set them all to not then return early?

Reply:
added